### PR TITLE
Refactor code into indexer-common crate

### PR DESCRIPTION
**Pull request by Jannis**
This moves out much of the common code into an `indexer-common` crate. The next step is likely to make more clear how one would go about implementing a different kind of indexer service.

### CODE REVIEW

**Comment by aasseman** [comment]
This is awesome, I actually just barely started working on this to share components with the "`indexer-tap-agent`" (or better name if someone feels inspired)

**Comment by Jannis** [comment]
@aasseman Ok, I expect this to pass the checks.

I didn't want to move the escrow monitor just yet. To avoid that, I had to split up the `test_vectors`. I think the TAP manager tests needed the allocation monitor and so I had to add a `mock` feature to the `indexer-common` crate to export the `faux` code generated for it across for testing. Probably something we can clean up again later.

**Comment by aasseman** [comment]
> @aasseman Ok, I expect this to pass the checks.
> 
> I didn't want to move the escrow monitor just yet. To avoid that, I had to split up the `test_vectors`. I think the TAP manager tests needed the allocation monitor and so I had to add a `mock` feature to the `indexer-common` crate to export the `faux` code generated for it across for testing. Probably something we can clean up again later.

I agree on all points. The mocking was what slowed me down considerably in my attempt to split things off into a common crate this weekend. I can take a stab at the EscrowMonitor in particular in another PR. I'm the only one depending on this for now so you don't need to struggle with this.

File: common/src/attestations/signers.rs [hunk 1]
```diff
@@ -21,7 +19,7 @@ pub struct AttestationSigners {
 }
 
 #[derive(Debug)]
-struct AttestationSignersInner {
+pub(crate) struct AttestationSignersInner {
```
Comment (aasseman) [inline common/src/attestations/signers.rs:22]: We don't need to make this public as of now.
```suggestion
struct AttestationSignersInner {
```

File: common/src/attestations/signers.rs [hunk 2]
```diff
@@ -55,7 +53,7 @@ impl AttestationSigners {
         }
     }
 
-    async fn update_attestation_signers(inner: Arc<AttestationSignersInner>) {
+    pub(crate) async fn update_attestation_signers(inner: Arc<AttestationSignersInner>) {
```
Comment (aasseman) [inline common/src/attestations/signers.rs:56]: Same as above
```suggestion
    async fn update_attestation_signers(inner: Arc<AttestationSignersInner>) {
```

**Review by aasseman — CHANGES_REQUESTED** [review]
LGTM except for 2 small things.
Also I think this should wait for #52 since this branch is based on top of it.

File: common/src/attestations/signers.rs (hunk 1 above)
Comment (Jannis) [reply common/src/attestations/signers.rs]: @aasseman You were right. Fixed!

File: common/src/attestations/signers.rs (hunk 2 above)
Comment (Jannis) [reply common/src/attestations/signers.rs]: Fixed in the latest commit.

**Comment by Jannis** [comment]
Rebased after #52 was merged. Let's see if the tests still pass.

File: service/src/server/routes/network.rs [hunk 3]
```diff
@@ -38,14 +39,25 @@ pub async fn network_queries(
...
-        _ => bad_request_response("Bad response from Graph node"),
+    if response.status().is_success() {
+        (
+            StatusCode::OK,
+            Json(
+                response
+                    .json::<Value>()
+                    .await
+                    // FIXME: Don't use expect here
+                    .expect("Failed to parse network subgraph query result"),
```
Comment (hopeyen) [inline service/src/server/routes/network.rs:56]: use another `bad_request_response`

**Review by aasseman — APPROVED** [review]

**Review by aasseman — APPROVED** [review]

**Pull request merged**